projects/ion/src/lib/tab-group/tab-group.component.ts

```diff
@@ -35,13 +53,33 @@ export class IonTabGroupComponent implements OnInit {
       vertical: 'right',
     };
 
-    if (this.border) {
+    if (this.isBorderDirectionCorrect(direction)) {
       return this.border;
     }
 
     return directions[direction] as BorderDirectionType;
   }
 
+  private getDirectionByBorder(border: BorderDirectionType): DirectionType {
+    const directions = {
+      left: 'vertical',
+      right: 'vertical',
+      top: 'horizontal',
+      bottom: 'horizontal',
+    };
+
+    return directions[border] as DirectionType;
```

Comment:
You can type directions object this way to remove the `as` assertion on return:
```suggestion
  private getDirectionByBorder(border: BorderDirectionType): DirectionType {
    const directions: {
      [key in BorderDirectionType]: DirectionType;
    } = {
      left: 'vertical',
      right: 'vertical',
      top: 'horizontal',
      bottom: 'horizontal',
    };

    return directions[border];
```